tritium-metrics/src/main/java/com/palantir/tritium/metrics/MetricRegistries.java

```diff
@@ -206,6 +209,29 @@ static void registerCache(MetricRegistry registry, Cache<?, ?> cache, String nam
                 .forEach((key, value) -> registerWithReplacement(registry, key, value));
     }
 
+    public static ScheduledExecutorService instrument(
+            TaggedMetricRegistry registry,
+            ScheduledExecutorService delegate,
+            String name) {
+        return new TaggedMetricsScheduledExecutorService(
+                checkNotNull(delegate, "delegate"),
+                checkNotNull(registry, "registry"),
+                checkNotNull(name, "name"));
```

Comment:
Previous implementation checked registry was not null twice, by using the results from checkNotNull we can leverage the compiler for validation.